app/plugin.go

```diff
@@ -425,19 +425,21 @@ func (a *App) GetMarketplacePlugins(filter *model.MarketplacePluginFilter) ([]*m
 		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_plugins.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
-	for _, plugin := range marketplacePlugins {
-		installedVersion := ""
+	for _, p := range marketplacePlugins {
+		marketplacePlugin := &model.MarketplacePlugin{
+			BaseMarketplacePlugin: p,
+		}
+
 		var manifest *model.Manifest
-		if manifest, err = pluginsEnvironment.GetManifest(plugin.Manifest.Id); err == nil {
+		if manifest, err = pluginsEnvironment.GetManifest(p.Manifest.Id); err != nil && err != plugin.ErrNotFound {
+			return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.config.app_error", nil, err.Error(), http.StatusInternalServerError)
+		} else if manifest != nil {
```

Comment:
nit: shouldn't we be testing `err == nil` here, as a parallel construction to `err != nil && err != plugin.ErrNotFound`? It seems like the `nil, nil` case is now unexpected, so it shouldn't be necessary to "test" using that approach?

Reply:
Ah yes, you are correct. Good catch!

Fixed.